src/coordinates/eop_manager.rs

```diff
@@ -101,46 +148,33 @@ impl EOPManager {
         }
 
         if self.eop_data.is_empty() {
-            return Err("No valid EOP data found in cache file".into());
+            return Err(EOPErrors::MissingEOPData);
         }
 
         Ok(())
     }
 
-    fn interpolate_eop_data(&self, epoch: Epoch) -> Result<EOPData, Box<dyn Error>> {
+    /// Interpolates EOP data for a given epoch.
+    fn interpolate_eop_data(&self, epoch: Epoch) -> Result<EOPData, EOPErrors> {
         if self.eop_data.is_empty() {
-            return Err("No EOP data available".into());
+            return Err(EOPErrors::MissingEOPData);
         }
 
-        // Convert Epoch to Unix timestamp
         let target_time = epoch.to_unix_seconds() as i64;
-
-        // Find the two closest data points
         let mut iter = self.eop_data.range(..=target_time);
         let after = iter.next_back();
         let before = iter.next_back();
 
         match (before, after) {
             (Some((&t1, eop1)), Some((&t2, eop2))) => {
-                // Calculate interpolation fraction
                 let fraction = (target_time - t1) as f64 / (t2 - t1) as f64;
                 Ok(EOPData::interpolate(eop1, eop2, fraction))
             }
             (Some((_, eop)), None) | (None, Some((_, eop))) => {
                 println!("Warning: Using nearest EOP value without interpolation");
                 Ok(eop.clone())
             }
-            (None, None) => {
-                println!("Warning: No valid EOP data found, using defaults");
-                Ok(EOPData {
-                    x_pole: 0.161556,
-                    y_pole: 0.247219,
-                    ut1_utc: -0.0890529,
-                    lod: 0.0017,
-                    ddpsi: -0.052,
-                    ddeps: -0.003,
-                })
-            }
```

Comment:
Here if there was an error we would just fall back to default data. Now we return an error.